astropy/wcs/wcsapi/wrappers/sliced_wcs.py

```diff
@@ -132,6 +151,39 @@ def __init__(self, wcs, slices):
             raise ValueError("Cannot slice WCS: the resulting WCS should have "
                              "at least one pixel and one world dimension.")
 
+
+    @property
+    def dropped_world_dimensions(self):
+        """
+        Information describing the dropped world dimensions.
+        """
+        world_coords = self._pixel_to_world_values_all(*[0]*len(self._pixel_keep))
+        dropped_info = defaultdict(list)
+
+        for i in range(self._wcs.world_n_dim):
+
+            if i in self._world_keep:
+                continue
+
+            if "world_axis_object_classes" not in dropped_info:
+                dropped_info["world_axis_object_classes"] = dict()
+
+            wao_classes = self._wcs.world_axis_object_classes
+            wao_components = self._wcs.world_axis_object_components
+
+            dropped_info["value"].append(world_coords[i])
+            dropped_info["world_axis_names"].append(self._wcs.world_axis_names[i])
+            dropped_info["world_axis_physical_types"].append(self._wcs.world_axis_physical_types[i])
+            dropped_info["world_axis_units"].append(self._wcs.world_axis_units[i])
+            dropped_info["world_axis_object_components"].append(wao_components[i])
+            dropped_info["world_axis_object_classes"].update(dict(
+                filter(
+                    lambda x: x[0] == wao_components[i][0], wao_classes.items()
+                )
+            ))
+            dropped_info["serialized_classes"] = self.serialized_classes
+        return dict(dropped_info)
```

Comment:
Is there a reason for doing this here rather than in the initializer? This will have to be computed every time ``dropped_world_dimensions`` is called? Alternatively we could use @lazyproperty?

Reply:
It seemed cleaner to do it in a property. Lazy property seems like a good idea. :+1: